storage/innobase/gis/gis0sea.cc

```diff
@@ -1572,17 +1530,16 @@ rtr_init_match(
 {
 	ut_ad(matches->matched_recs->empty());
 	matches->locked = false;
-	rtr_copy_buf(matches, block);
-	matches->block.frame = matches->bufp;
 	matches->valid = false;
-	/* We have to copy PAGE_W*_SUPREMUM_END bytes so that we can
+	if (!matches->block) {
+		matches->block = buf_block_alloc();
+	}
+	/* We have to copy PAGE_*_SUPREMUM_END bytes so that we can
 	use infimum/supremum of this page as normal btr page for search. */
-	memcpy(matches->block.frame, page, page_is_comp(page)
-						? PAGE_NEW_SUPREMUM_END
-						: PAGE_OLD_SUPREMUM_END);
 	matches->used = page_is_comp(page)
 				? PAGE_NEW_SUPREMUM_END
 				: PAGE_OLD_SUPREMUM_END;
+	memcpy(matches->block->frame, page, matches->used);
```

Comment:
You are right, we do use the page ID, and it needs to be set.

I did see an occasional failure of the test `innodb_gis.rtree_compress` while testing a work-in-progress version of MDEV-29445, which depends on this cleanup. Many tests in that suite are disabled because they very often fail. I will run all tests in that suite to check this.

Reply:
2b30a8587e23dbc52feb46d766ef59902071aeaf addresses your comment and #3185 addresses mine.